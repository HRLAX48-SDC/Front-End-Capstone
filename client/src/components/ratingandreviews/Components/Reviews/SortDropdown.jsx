import React, { useState, useEffect} from 'react';
import Dropdown from 'react-bootstrap/Dropdown'
import DropdownButton from 'react-bootstrap/DropdownButton'


function SortDropdown (props) {

  //make the tile of the button change to what item was clicked on
  const[title, setTitle]= useState('Relevance');
  
  useEffect(() => {
    props.sortBy(title)
  }, [title])
  
  const changeTitle = (e) =>{
    setTitle(e.target.name)
    props.sortBy(title)
  } 
  // const sortBy = () => {
  //   props.sortBy(title)
  // }


    return (
      <DropdownButton id="dropdown-basic-button" title={title} variant={'secondary'} >
        <Dropdown.Item href="#/action-1" name="Relevant" onClick={changeTitle}>Relevant</Dropdown.Item>
        <Dropdown.Item href="#/action-2" name="Newest" onClick={changeTitle}>Newest</Dropdown.Item>
        <Dropdown.Item href="#/action-3" name="Helpful" onClick={changeTitle}>Helpful</Dropdown.Item>
      </DropdownButton>
    )
}
export default SortDropdown;

