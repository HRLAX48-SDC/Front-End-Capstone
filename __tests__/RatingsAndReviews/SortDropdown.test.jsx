import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import SortDropdown from '../../client/src/components/ratingandreviews/Components/Reviews/SortDropdown.jsx'

describe('<SortDropdown />', () => {
    const sortBy = () => {}
      it('Should inherit a sort by method', () => {
          const sortDropdown = renderer
          .create(<SortDropdown sortBy={sortBy}/>)
          .toJSON();
          expect(sortDropdown).toMatchSnapshot()
        });
        
        // it('Should have an initial state of relevant', () => {
    
        // })
});