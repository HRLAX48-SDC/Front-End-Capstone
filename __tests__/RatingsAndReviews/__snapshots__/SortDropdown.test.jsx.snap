// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SortDropdown /> Should inherit a sort by method 1`] = `
<div
  className="dropdown"
>
  <button
    aria-expanded={false}
    className="dropdown-toggle btn btn-outline-secondary"
    disabled={false}
    id="dropdown-basic-button"
    onClick={[Function]}
    type="button"
  >
    relevant
  </button>
</div>
`;
